src/java.base/share/classes/java/util/Properties.java

```diff
@@ -165,6 +172,42 @@
      */
     private transient volatile ConcurrentHashMap<Object, Object> map;
 
+    @SuppressWarnings("removal")
+    private static final class LazyDateCommentProvider {
+        // format that matches the one used by java.util.Date.toString()
+        private static final String dateFormatPattern = "EEE MMM dd HH:mm:ss zzz yyyy";
+        // formatter used while writing out current date
+        private static final DateTimeFormatter currentDateFormatter = DateTimeFormatter.ofPattern(dateFormatPattern);
+        private static final String cachedDateComment;
+
+        static {
+            String sourceDateEpoch = System.getSecurityManager() == null
+                                        ? System.getenv("SOURCE_DATE_EPOCH")
+                                        : AccessController.doPrivileged((PrivilegedAction<String>)
+                                                () -> System.getenv("SOURCE_DATE_EPOCH"));
```

Comment:
The format of SOURCE_DATE_EPOCH is fine, but referring to an external document as part of the OpenJDK spec is undesirable in the long run. Previous comments gave support to using the environment variable directly but
it very unlike similar cases that use system properties when the behavior of an API needs to be overridden or modified.  

As a system property, specified on the command line, it would be visible when the program is invoked,
explicitly intended to change behavior and not having a context sensitive effect.  Named perhaps, java.util.Properties.storeDate.

It can be documented as part of the spec with the javadoc tag "{@systemProperty  ...}"
There is a cache of properties in jdk.internal.util.StaticProperty that captures the state when the runtime is initialized. For specific properties, they are cached and available without regard to the setting of the SecurityManager or not.  (BTW, there is no need to special case doPriv calls, they are pretty well optimized when the security manager is not set and it keeps the code simpler.)

Given the low frequency of calls to store(), even caching the parsed date doesn't save much.
And the cost is the cost of the size and loading an extra class.